context.go

```diff
@@ -643,6 +646,25 @@ func checkMissingPositionals(positional int, values []*Value) error {
 	return fmt.Errorf("missing positional arguments %s", strings.Join(missing, " "))
 }
 
+func checkXorDuplicates(paths []*Path) error {
+	seen := map[string]*Flag{}
+	for _, path := range paths {
+		for _, flag := range path.Flags {
```

Comment:
This looks like it will xor across subcommands. I'm not necessarily against that, but is that preferable to xor within a single scope? Seems like you could get collisions when composing hierarchies of flags/commands.

Reply:
I did consider both. The hierarchical version seems better, as long as the users use a different group name. I'll update the README.

Reply:
Actually can you expand vars in this attribute? That will allow
For namespacong.

Reply:
Actually can you expand vars in this attribute? That will allow
For namespacong.

Reply:
As discussed offline, I think a non-hierarchical `xor` is a better idea. If you're composing flags from a bunch of structs, in a large application, it's not unlikely that you will have colliding xor groups, but there will be no indication that it will collide, until a user attempts to use both sets of flags.

It also makes composition/reuse more difficult, eg. say you have something like this:

```go
type Config struct {
  File string `xor:"source" help:"Configuration file."`
  JSON string `xor:"source" help:"JSON configuration blob."`
}
```

But you want to reuse this in multiple places, namespaced via a prefix:

```go
type CLI struct {
  BaseConfig Config `embed:"" prefix"base-"`
  ChildConfig Config `embed:"" prefix"child-"`
}
```

The xor group will conflict, making reuse impossible.